e2e-tests/cypress/support/page-objects/Spaces.ts

```diff
@@ -1,10 +1,24 @@
+import * as semver from 'semver';
+import { getKibanaVersion } from '../helpers';
+
 export class Spaces {
   static removeSpace(spaceName: string) {
     cy.log('Remove space');
+    const spaceNameLowerCaseAndDash = spaceName.toLowerCase().replace(' ', '-');
 
     cy.get('[data-test-subj=spacesNavSelector]').click();
     cy.get('[data-test-subj=manageSpaces]').click({ force: true });
-    cy.get(`[data-test-subj="${spaceName} space-deleteSpace"]`).click({ force: true });
+    if (semver.gte(getKibanaVersion(), '8.16.0')) {
+      cy.get(`[id="${spaceNameLowerCaseAndDash}-actions"]`).click();
+      cy.get(`[data-test-subj="${spaceNameLowerCaseAndDash}-deleteSpace"]`).click();
+    } else {
+      cy.get(`[data-test-subj="${spaceName}-deleteSpace"]`).click();
+    }
+
+    cy.intercept(`s/${spaceNameLowerCaseAndDash}/api/spaces/space/${spaceNameLowerCaseAndDash}`).as('deleteSpace');
```

Comment:
not only for 8.16.0?

Reply:
No, it's valid for all versions, so after deleting, we will wait for ES's 204 status to detect if the delete request was successful.